src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -634,5 +1083,119 @@ internal bool TryGetSpanOnHigherBuffer(SnapshotSpan snapshotSpan, ITextBuffer ta
             span = default;
             return false;
         }
+
+        private sealed class State
+        {
+            /// <summary>
+            /// The current expansion session.
+            /// </summary>
+            public IVsExpansionSession _expansionSession;
+
+            /// <summary>
+            /// The set of symbols initially identified as candidates for providing arguments. When a snippet is
+            /// constructed with parameters for a specific symbol, <see cref="_method"/> will identify the specific
+            /// symbol from within this collection matching the current session.
+            /// </summary>
+            /// <remarks>
+            /// <para>This collection might not contain <see cref="_method"/>, particularly in cases where
+            /// <see cref="GetReferencedSymbolsToLeftOfCaretAsync"/> returns only a subset of the available overloads.
+            /// One simple case can be seen in Visual Basic code invoking <see cref="int.ToString()"/>:</para>
+            ///
+            /// <code>
+            /// Dim x = 0
+            /// x.ToString$$
+            /// </code>
+            ///
+            /// <para>When <see cref="GetReferencedSymbolsToLeftOfCaretAsync"/> is invoked at the caret location,
+            /// <see cref="int.ToString()"/> is returned, but other overloads like <see cref="int.ToString(string)"/>
+            /// are not. This is due to the fact that parentheses are optional for invocations that do not have any
+            /// parameters, as opposed to the equivalent C# case where <c>ToString</c> would refer to a method
+            /// group.</para>
+            /// </remarks>
+            public ImmutableArray<IMethodSymbol> _methods;
```

Comment:
So I don't really see how _methods is being used here -- I see it being used in IsFullMethodCallSnippet but it seems like can just be a boolean state of whether we're in this mode or not?

Reply:
➡️ It probably could be changed, but this is going to have to change anyway as part of nested call support so just leaving it for now. Changes in this space have caused undesired "iteration" churn (subtle fallout) that I don't want to go through again since the current form works.

Reply:
Oh I think I see the bit I might be missing: if you do tab + tab on a method name we'll stick in parenthesis, we stick the methods here to recognize the name later. And this line:

    var symbolName = _state._method?.Name ?? _state._methods.FirstOrDefault()?.Name;

Is just doing FirstOrDefault() because all the methods in the group have the same name? Yeah, this is really confusing. :frowning: